sdk/src/main/java/io/radar/sdk/RadarAbstractLocationClient.kt

```diff
@@ -0,0 +1,57 @@
+package io.radar.sdk
+
+import android.app.PendingIntent
+import android.location.Location
+
+internal abstract class RadarAbstractLocationClient() {
+
+    data class RadarAbstractGeofence(
+        val requestId: String,
+        val latitude: Double,
+        val longitude: Double,
+        val radius: Float,
+        val transitionEnter: Boolean = false,
+        val transitionExit: Boolean = false,
+        val transitionDwell: Boolean = false,
+        val dwellDuration: Long = 0,
+        val loiteringDelay: Int = 0
+    )
+
+    data class RadarAbstractGeofenceRequest(
+        val initialTriggerEnter: Boolean = false,
+        val initialTriggerExit: Boolean = false,
+        val initialTriggerDwell: Boolean = false
+    )
+
+    abstract fun getCurrentLocation(
+        desiredAccuracy: RadarTrackingOptions.RadarTrackingOptionsDesiredAccuracy,
+        block: (location: Location?) -> Unit
+    )
+
+    abstract fun requestLocationUpdates(
+        desiredAccuracy: RadarTrackingOptions.RadarTrackingOptionsDesiredAccuracy,
+        interval: Int,
+        fastestInterval: Int,
+        pendingIntent: PendingIntent
+    )
+
+    abstract fun removeLocationUpdates(
+        pendingIntent: PendingIntent
+    )
+
+    abstract fun getLastLocation(
+        block: (location: Location?) -> Unit
```

Comment:
block -> handler or locationHandler ?

Reply:
Not sure I follow. What do you mean?

Reply:
Ah, got it. Just following the same naming convention as lambda callbacks in the `Radar` class.

When this is called, the code looks like

```
locationClient.getLastLocation { location: Location? ->
  // ...
}
```

so it doesn't matter too much anyway.

Reply:
works for me